client/my-sites/activity/activity-log/threat-alert.jsx

```diff
@@ -150,12 +341,13 @@ export class ThreatAlert extends Component {
 											dateFormat="ll"
 										/>
 									</span>
+									{ this.getSpinner() }
 									<SplitButton
 										compact
 										primary
 										label={ translate( 'Fix threat' ) }
-										onClick={ () => debug( 'main button clicked' ) }
-										disabled={ false }
+										onClick={ this.fixThreat }
```

Comment:
Not all threats are fixable. Not sure what we want to do design-wise in that case though; can we disable the main button but still have the pop-over? Or should `Ignore` move into the main button area?

It'd also be nice to use the URL for fixing that's passed in via the API, which I think you're now getting with the `threat.actions` mapping?

Reply:
> use the URL for fixing that's passed in via the API

I think we found that that URL was wrong, possibly the namespace was wrong. Anyway it 404'd.

Reply:
Ah, so that should be fixed. It looks like `$this->rest_base` is in the wrong place in the URL; `site-alerts.php:get_site_alerts()`. The presence of that action is one indication that the threat can be fixed, although I believe we also have the `fixable` flag in the threat data too.

Reply:
> that should be fixed

I'm not totally sold on the design of having the URL in the response, given that we can construct it (correctly) anyway, making it redundant.

Reply:
>making it redundant.
I agree 1000%. Also, other requests everywhere on Calypso are built the same way as I've done it here, so using the URL from a service deviates the implementation from the main design.

Reply:
>I think we found that that URL was wrong, possibly the namespace was wrong. Anyway it 404'd.

The URL is fine. It's just that the wpcom-http::http() method is quirky. I'll probably make a PR for that too.

Reply:
>Not all threats are fixable. 

@singerb , could you please give me an example of a non-fixable threat?

Reply:
As I said, I'm not wedded to doing it this way, especially if it's the only place in Calypso that does.